tools/release.sh

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env bash
+
+# Script to make release on the local machine.
+# See https://github.com/realm/realm-wiki/wiki/Java-Release-Checklist for more details.
+# FIXME: Only patch release is supported now.
+
+set -euo pipefail
+IFS=$'\n\t'
+
+usage() {
+cat <<EOF
+Usage: $0 <branch-to-release>
+EOF
+}
+
+######################################
+# Input Validation
+######################################
+
+if [ "$#" -eq 0 ] || [ "$#" -gt 1 ] ; then
+    usage
+    exit 1
+fi
+
+######################################
+# Variables
+######################################
+
+BRANCH_TO_RELEASE="$1"
+VERSION=""
+REALM_IO_PATH=${REALM_IO_PATH:-}
+
+check_adb_device() {
+    if ! adb get-state 1>/dev/null 2>&1 ; then
+        read -n 1 -s -p -r "Attach a test device or start the emulator then press any key to continue..."
+        echo ""
+        check_adb_device
+    fi
+}
+
+check_env() {
+    echo "Checking environment..."
+
+    # Try to find s3cmd
+    path_to_s3cmd=$(which s3cmd)
+    if [[ ! -x "$path_to_s3cmd" ]] ; then
+        echo "Cannot find executable file 's3cmd'."
+        exit -1
+    fi
+    if [[ ! -e "$HOME/.s3cfg" ]] ; then
+        echo "'$HOME/.s3cfg' cannot be found."
+        exit -1
+    fi
+
+    # Check BinTray credentials
+    if ! grep "bintrayUser=realm" "$HOME/.gradle/gradle.properties" > /dev/null ; then
+        echo "'bintrayUser' is not set in the '$HOME/.gradle/gradle.properties'."
+        exit -1
+    fi
+
+    if ! grep "bintrayKey=.*" "$HOME/.gradle/gradle.properties" > /dev/null; then
+        echo "'bintrayKey' is not set in the '$HOME/.gradle/gradle.properties'."
+        exit -1
+    fi
+
+    # Check gradle params
+    if grep buildTargetABIs "$HOME/.gradle/gradle.properties" | grep -v "^#" > /dev/null ; then
+        echo "'buildTargetABIs' should be disabled in the '$HOME/.gradle/gradle.properties'."
+        exit -1
+    fi
+    if grep ccachePath "$HOME/.gradle/gradle.properties" | grep -v "^#" > /dev/null ; then
+        echo "'ccachePath' should be disabled in the '$HOME/.gradle/gradle.properties'."
+        exit -1
+    fi
+    if grep lcachePath "$HOME/.gradle/gradle.properties" | grep -v "^#" > /dev/null ; then
+        echo "'lcachePath' should be disabled in the '$HOME/.gradle/gradle.properties'."
+        exit -1
+    fi
+
+    if [[ -z ${REALM_IO_PATH} ]] ; then
+        REALM_IO_PATH="$(pwd)/../realm.io"
+    fi
+    if [[ ! -e ${REALM_IO_PATH} ]] ; then
+        echo "Please set 'REALM_IO_PATH' to the 'realm.io' repository path to publish javadoc."
+        exit -1
+    fi
+}
+
+prepare_branch() {
+    echo "Preparing release branch..."
+
+    git fetch --all
+    git checkout releases
+    git reset --hard origin/releases
+    git clean -xfd
```

Comment:
I think so. And I really hope people release from a dedicated project instead of the daily working project directory. Make sure the project dir is clean is very much the most important thing for release.